### ISSUE TITLE
Cannot connect to any of my servers that have installations of code-server on ubuntu anymore.

### ISSUE THREAD

**Issue by escozul** [microsoft/vscode#185106]
I started getting this error yesterday.
I'm unable to connect using tunnel connection to my vscode servers through a browser.

Using SSH from the client still works, but the tunnel that I use when I don't have a locally install vscode client doesn't.

Can't figure out why. I only have an "Opening Remote" notification at the lower left of my window, which is normal, but it should connect after a while. On the server side, the code-server starts without warnings and wakes up properly.

It just never finishes initializing and I need to close the connection after a while.

Version: 1.79.1
Commit: 4cb974a7aed77a74c7813bdccd99ee0d04901215
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
Embedder: vscode.dev

**Comment by connor4312** [comment]
Are you still using the `code-server` CLI? Back in December we released the [new `code` CLI](https://code.visualstudio.com/docs/remote/tunnels#_using-the-code-cli) which includes tunnels and is the recommended way to connect.

If so, please grab two things:

1. Run the tunnel via `code tunnel --verbose` on the server side, and grab logs from when you try to connect
2. Check for any console errors on vscode.dev

Thanks!

**Comment by ElanHasson** [comment]
I have the same issue, using SSH started yesterday after upgrade. VSCode Insiders build works fine. Preview release of SSH extension with Stable does not work.

**Comment by escozul** [comment]
> Are you still using the `code-server` CLI? Back in December we released the [new `code` CLI](https://code.visualstudio.com/docs/remote/tunnels#_using-the-code-cli) which includes tunnels and is the recommended way to connect.
> 
> If so, please grab two things:
> 
> 1. Run the tunnel via `code tunnel --verbose` on the server side, and grab logs from when you try to connect
> 2. Check for any console errors on vscode.dev
> 
> Thanks!

Hey that is... I was looking for updates on that and I did find it annoying that it kept wartning me about the old version of my tunnel but when I was checking, there was no update.

However this code-server installation is setup on an ubuntu headless machine which I want to be able to connect to through the web. That code cli version is windows only... or am I missing something... I trully think I'm missing something like a procedure to properly install it using ssh on my ubuntu machine? No GUI on that machine either. it has to be terminal based and only available through tunnel and ssh.

**Comment by connor4312** [comment]
The VS Code CLI si available on all platforms and does not require a GUI. You can get it under the "CLI" heading on all platforms on the downloads page: https://code.visualstudio.com/#alt-downloads

**Comment by escozul** [comment]
And it's not that right: https://github.com/coder/code-server

Sorry for being so persistent man, but everyone keeps referring to each version as vscode, and code-server and code cli and vscode server... and I'm not sure if these are just similarly named DIFFERENT iterations or simply the one and only vscode.

**Comment by connor4312** [comment]
That project is not something the vs code team has anything to do with. If you're having a problem with it, please open an issue on their repo.

**Comment by escozul** [comment]
> That project is not something the vs code team has anything to do with. If you're having a problem with it, please open an issue on their repo.

I am not using that project. I just asked if it was the same or not.
In any case this: https://code.visualstudio.com/blogs/2022/07/07/vscode-server is related to a project from the vs code team.
This is the one I'm using now and I'm looking for a way to use the latest version. So How will I install this: https://code.visualstudio.com/#alt-downloads on an ubuntu server? The vscode-server project had an install.sh file. The code cli project doesn't have one... or I can't find it.

**Comment by connor4312** [comment]
ah, sorry, misunderstood.

To install that, you would download it directly, uncompress it, and run the CLI. We have a todo to add a nice rustup.rs-like experience for the website.

```
curl https://code.visualstudio.com/sha/download?build=stable&os=cli-alpine-x64 -o vscode-cli.tar.gz
tar -xvf ./vscode-cli.tar.gz
./code tunnel
```

**Comment by escozul** [comment]
> ah, sorry, misunderstood.
> 
> To install that, you would download it directly, uncompress it, and run the CLI. We have a todo to add a nice rustup.rs-like experience for the website.
> 
> ```
> curl https://code.visualstudio.com/sha/download?build=stable&os=cli-alpine-x64 -o vscode-cli.tar.gz
> tar -xvf ./vscode-cli.tar.gz
> ./code tunnel
> ```

Ok Then. So it’s manual installation in my case I guess.
I saw the download tar and run `./code tunnel` part.
I can add here that if you move the executable `code` to \usr\local\bin` (which should also be in your PATH) you won’t need to run `./code` commands but straight up `code` from every location. Makes the experience much like having it installed.

That little thing might make a useful addition to that article here that describes how to set up a tunnel: https://code.visualstudio.com/docs/remote/tunnels#_using-the-code-cli

Additionally, I created a run_server and stop_server sh script, aliased them, and have those to start and stop the developing environment. Other people might prefer to have them auto-start or something.

I'm really happy to have my web development environment back!
